pg_replicate/src/clients/postgres.rs

```diff
@@ -136,13 +136,37 @@ impl ReplicationClient {
         Ok(stream)
     }
 
-    /// Returns a vector of columns of a table
+    /// Returns a vector of columns of a table, optionally filtered by a publication's column list
     pub async fn get_column_schemas(
         &self,
         table_id: TableId,
+        publication: Option<&str>,
     ) -> Result<Vec<ColumnSchema>, ReplicationClientError> {
+        let (pub_cte, pub_pred) = if let Some(publication) = publication {
+            (
+                format!(
+                    "with pub_attrs as (
+                        select unnest(r.prattrs)
+                        from pg_publication_rel r
+                        left join pg_publication p on r.prpubid = p.oid
+                        where p.pubname = '{publication}'
```

Comment:
`publication` should be run through the `quote_literal` function to avoid sql injection.

Reply:
Sure. Actually, would you rather this or to change `simple_query` to `query` and use bindings?

Reply:
IIRC `query` won't work here because a replication connection only supports the [simple query protocol](https://www.postgresql.org/docs/current/protocol-flow.html#PROTOCOL-FLOW-SIMPLE-QUERY)

Reply:
Aha, that's a great insight, thanks.